nimbus-core/src/main/java/com/antheminc/oss/nimbus/domain/model/state/extension/ScriptEventHandler.java

```diff
@@ -46,7 +52,7 @@
  *
  */
 @Getter
-public class ScriptStateLoadNewHandler implements OnStateLoadNewHandler<Script> {
+public class ScriptEventHandler implements OnStateLoadNewHandler<Script>, OnStateChangeHandler<Script>, OnStateLoadHandler<Script> {
```

Comment:
Given that an annotation could be lazy marked for a type of event to handle, I think we need to come up with a base EventHandler, which mandates the exact event type to be made available either via eventType attribute on the annotation or derived from meta inherited annotation. This would remove the need to upfront implement all types of possible handlers and also avoid updating this class if we add more event types in future.

Reply:
so, right now we have separate interfaces with one method (with diff. parameters) specific to that interface 
e.g.
OnStateLoadHandler interface-> onStateLoad( annotation, param), 
OnStateChangeHandler -> onStateChange(annotation, txnCtx, paramEvent)
OnTxnExecuteHandler -> handleOnStart(annotation, txnCtx)
...

so if we go with single EventHandler interface, do we create default methods (default throwing exception if not overridden) 
e.g. 
onEvent(annotation, param) 
onEvent(annotation. txnCtx, paramEvent)
onEvent(annotation. txnCtx) 

and then call the appropriate methods where applicable during entity load and change ?

Reply:
I think the objectives i.e., common abstraction for such handlers to allow reuse of common code, minimizing changes due to future event types/handler being added - could be handled by adding a new AbstractEventHandlerSupport class which can implement all the known event handlers currently. 

The default implementation provided for these methods can be an InvalidConfigException (or similar) suggesting that not all event types are supported (same as what you mentioned). The annotation documentation should also call out which ones are supported by creating an Enum.Subset to enforce it.

Reply:
Updated the changes. please review.